com.unity.render-pipelines.universal/Editor/Camera/UniversalRenderPipelineCameraEditor.cs

```diff
@@ -637,7 +640,41 @@ void DrawEnvironmentSettings(CameraRenderType camType)
                         }
                     }
                 }
-                DrawVolumes();
+
+                EditorGUILayout.Space();
+                EditorGUILayout.Space();
+            }
+            EditorGUILayout.EndFoldoutHeaderGroup();
+        }
+
+        void DrawVolumeSettings()
+        {
+            m_VolumesSettingsFoldout.value = EditorGUILayout.BeginFoldoutHeaderGroup(m_VolumesSettingsFoldout.value, Styles.volumesSettingsText);
+            if (m_VolumesSettingsFoldout.value)
+            {
+                // Display the Volume Update mode, LayerMask and Trigger
+                LayerMask selectedVolumeLayerMask = m_SerializedCamera.volumeLayerMask.intValue;
+                Transform selectedVolumeTrigger = (Transform)m_SerializedCamera.volumeTrigger.objectReferenceValue;
+
+                EditorGUI.BeginChangeCheck();
+
+                EditorGUILayout.PropertyField(m_SerializedCamera.volumeFrameworkUpdateMode, Styles.volumeUpdates);
+                DrawLayerMask(m_SerializedCamera.volumeLayerMask, ref selectedVolumeLayerMask, Styles.volumeLayerMask);
+                DrawObjectField(m_SerializedCamera.volumeTrigger, ref selectedVolumeTrigger, Styles.volumeTrigger);
+
+                if (EditorGUI.EndChangeCheck())
+                {
+                    m_SerializedCamera.volumeLayerMask.intValue = selectedVolumeLayerMask;
+                    m_SerializedCamera.volumeTrigger.objectReferenceValue = selectedVolumeTrigger;
+                    m_SerializedCamera.Apply();
+
+                    Camera cam = (Camera)m_SerializedCamera.serializedObject.targetObject;
+                    if (cam != null)
+                    {
+                        cam.UpdateVolumeStack();
+                    }
```

Comment:
Could we do this in `AddionalCameraData.OnValidate`? Any benefits to doing this here?

Reply:
The benefits I see are that we only call the update function when we change anything regarding volumes. If I move that to OnValidate() then it will be triggered by any change done in the UI.